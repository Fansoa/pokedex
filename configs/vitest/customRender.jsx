import { render } from "@testing-library/react";
import IntegrationProvider from "@/configs/vitest/testUtils";

const renderWithIntegrationProvider = (children) =>
  render(<IntegrationProvider>{children}</IntegrationProvider>);

export default renderWithIntegrationProvider;
