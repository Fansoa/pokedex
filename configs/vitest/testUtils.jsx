import { ThemeProvider } from "@emotion/react";
import PropTypes from "prop-types";
import theme from "@/src/theme/theme";

const IntegrationProvider = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

IntegrationProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default IntegrationProvider;
