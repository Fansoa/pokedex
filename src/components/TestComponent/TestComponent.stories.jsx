import TestComponent from "@/src/components/TestComponent";

export default {
  title: "Components/TestComponent",
  component: TestComponent,
  args: {
    variant: "bleu",
  },
};

export const TestComponentStory = {
  name: "Default",
};
