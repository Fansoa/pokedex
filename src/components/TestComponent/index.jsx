import PropTypes from "prop-types";
import { useState } from "react";
import { useTranslation } from "react-i18next";
import Container from "./styled";

const TestComponent = ({ variant }) => {
  const [isDisplay, setIsDisplay] = useState(false);

  const { t, i18n } = useTranslation();

  const languagesOptions = [
    { label: t("glossary:english"), value: "en" },
    { label: t("glossary:french"), value: "fr" },
  ];

  return (
    <Container variant={variant}>
      <p>TestComponent</p>
      <button
        type="button"
        onClick={() => {
          setIsDisplay((curr) => !curr);
        }}
      >
        {t("common:display.the.text")}
      </button>
      <select
        onChange={(e) => i18n.changeLanguage(e.target.value)}
        value={i18n.language}
      >
        {languagesOptions.map(({ label, value }) => (
          <option key={`language-option-${label}`} value={value}>
            {label}
          </option>
        ))}
      </select>
      {isDisplay && <p>Lorem Ipsum</p>}

      {t("glossary:current.language")}
    </Container>
  );
};

TestComponent.propTypes = {
  variant: PropTypes.oneOf(["bleu", "rouge"]),
};

TestComponent.defaultProps = {
  variant: "bleu",
};

export default TestComponent;
