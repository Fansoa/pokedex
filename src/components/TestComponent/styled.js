import styled from "@emotion/styled";

const Container = styled.div`
  ${({ theme, variant }) => `
    background-color: ${theme.testComponentTheme[variant].backgroundColor}
  `}
`;

export default Container;
