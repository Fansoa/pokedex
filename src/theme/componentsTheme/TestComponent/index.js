import global from "@/src/theme/global";

const testComponentTheme = {
  bleu: {
    backgroundColor: global.colors.blue,
  },
  rouge: {
    backgroundColor: global.colors.red,
  },
};

export default testComponentTheme;
