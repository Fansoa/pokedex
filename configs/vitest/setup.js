import { expect, afterEach } from "vitest";
import { cleanup } from "@testing-library/react";
import * as matchers from "@testing-library/jest-dom/matchers";
import { createSerializer } from "@emotion/jest";

expect.extend(matchers);
expect.addSnapshotSerializer(createSerializer());

afterEach(() => {
  cleanup();
});
