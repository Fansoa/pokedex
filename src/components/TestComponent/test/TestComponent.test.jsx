import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { expect } from "vitest";
import renderWithIntegrationProvider from "@/configs/vitest/customRender";
import TestComponent from "@/src/components/TestComponent";
import { act } from "react-dom/test-utils";

describe("TestComponent", () => {
  let container;

  it("Label is displayed", () => {
    container = renderWithIntegrationProvider(<TestComponent />).container;
    const label = screen.getByText(/testcomponent/i);
    expect(label).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("isClicked is displayed when button is clicked", async () => {
    container = renderWithIntegrationProvider(<TestComponent />).container;
    const user = userEvent.setup();
    const button = screen.getByRole("button", {
      name: /common:display.the.text/i,
    });
    await act(async () => {
      await user.click(button);
    });
    const isClicked = screen.getByText(/lorem ipsum/i);
    expect(isClicked).toBeInTheDocument();
    await act(async () => {
      await user.click(button);
    });
    expect(isClicked).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
