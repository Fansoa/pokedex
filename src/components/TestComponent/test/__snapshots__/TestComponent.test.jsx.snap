// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TestComponent > Label is displayed 1`] = `
.emotion-0 {
  background-color: blue;
}

<div>
  <div
    class="emotion-0"
  >
    <p>
      TestComponent
    </p>
    <button
      type="button"
    >
      common:display.the.text
    </button>
    <select>
      <option
        value="en"
      >
        glossary:english
      </option>
      <option
        value="fr"
      >
        glossary:french
      </option>
    </select>
    glossary:current.language
  </div>
</div>
`;

exports[`TestComponent > isClicked is displayed when button is clicked 1`] = `
.emotion-0 {
  background-color: blue;
}

<div>
  <div
    class="emotion-0"
  >
    <p>
      TestComponent
    </p>
    <button
      type="button"
    >
      common:display.the.text
    </button>
    <select>
      <option
        value="en"
      >
        glossary:english
      </option>
      <option
        value="fr"
      >
        glossary:french
      </option>
    </select>
    glossary:current.language
  </div>
</div>
`;
